src/codemodder/executor.py

```diff
@@ -0,0 +1,88 @@
+from importlib.abc import Traversable
+
+from wrapt import CallableObjectProxy
+
+
+class CodemodExecutorWrapper(CallableObjectProxy):
+    """A wrapper around a codemod that provides additional metadata."""
+
+    origin: str
+    docs_module: Traversable
+    semgrep_config_module: Traversable
+
+    def __init__(
+        self,
+        codemod,
+        origin: str,
+        docs_module: Traversable,
+        semgrep_config_module: Traversable,
+    ):
+        super().__init__(codemod)
+        self.origin = origin
+        self.docs_module = docs_module
+        self.semgrep_config_module = semgrep_config_module
+
+    def apply(self, context):
+        """
+        Wraps the codemod's apply method to inject additional arguments.
+
+        Not all codemods will need these arguments.
+        """
+        return self.apply_rule(context, yaml_files=self.yaml_files)
```

Comment:
should this not be `codemod.apply_rule`? I don't see an `apply_rule` definition in the executor class

Reply:
This is because I've implemented it as a proxy so it automatically delegates to the wrapped object's methods.

Reply:
ohhhhhhh magic